test/ASTGen/decls.swift

```diff
@@ -105,6 +105,11 @@ func testVars() {
   }
 }
 
+func rethrowingFn(fn: () throws -> Void) rethrows {}
+func testRethrows() {
+    rethrowingFn { _ = 1 }
+}
+
```

Comment:
Worth adding a test for `reasync`? Interestingly, `-enable-experimental-concurrency` doesn't require an asserts compiler